### PR TITLE
Upmerge PR#83 to dev_ae_stdxl_ft

### PR DESCRIPTION
**Pull request by yeonsily**
# What does this PR do?

Fixes # (issue)

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you make sure to update the documentation with your changes?
- [ ] Did you write any new necessary tests?

### CODE REVIEW

File: examples/stable-diffusion/training/train_text_to_image_sdxl.py [hunk 1]
```diff
@@ -66,14 +57,15 @@
...
+from optimum.habana.utils import set_seed, HabanaProfile
+from optimum.habana.accelerate.utils.dataclasses import GaudiDistributedType
+from optimum.habana.utils import to_gb_rounded
 
 if is_wandb_available():
     import wandb
 
 # Will error if the minimal version of diffusers is not installed. Remove at your own risks.
-check_min_version("0.26.0")
+check_min_version("0.23.1")
```
Comment (libinta) [inline examples/stable-diffusion/training/train_text_to_image_sdxl.py:68]: can we update this?

File: examples/stable-diffusion/training/train_text_to_image_sdxl.py [hunk 2]
```diff
@@ -1032,162 +918,54 @@ def collate_fn(examples):
...
 
-        # Move unet, vae and text_encoder to device and cast to weight_dtype
-        # The VAE is in float32 to avoid NaN losses.
-        vae.to(accelerator.device, dtype=weight_dtype)
-        text_encoder_one.to(accelerator.device, dtype=weight_dtype)
-        text_encoder_two.to(accelerator.device, dtype=weight_dtype)
-        del text_encoders, tokenizers
-        gc.collect()
-        # Create EMA for the unet.
-        if args.use_ema:
```
Comment (libinta) [inline examples/stable-diffusion/training/train_text_to_image_sdxl.py:1151]: should we enable 1151-1189?

File: examples/stable-diffusion/training/train_text_to_image_sdxl.py [hunk 3]
```diff
@@ -1279,28 +1062,20 @@ def unwrap_model(model):
         disable=not accelerator.is_local_main_process,
     )
 
+    import habana_frameworks.torch as htorch
```
Comment (libinta) [inline examples/stable-diffusion/training/train_text_to_image_sdxl.py:1065]: move this to top

File: examples/stable-diffusion/training/train_text_to_image_sdxl.py [hunk 4]
```diff
@@ -1411,34 +1168,34 @@ def compute_time_ids(original_size, crops_coords_top_left):
...
             # Checks if the accelerator has performed an optimization step behind the scenes
             if accelerator.sync_gradients:
                 progress_bar.update(1)
                 global_step += 1
-                accelerator.log({"train_loss": train_loss}, step=global_step)
-                train_loss = 0.0
 
                 if accelerator.is_main_process:
-                    if args.checkpointing_steps is not None and global_step % args.checkpointing_steps == 0:
+                    if global_step % args.checkpointing_steps == 0:
```
Comment (libinta) [inline examples/stable-diffusion/training/train_text_to_image_sdxl.py:1198]: should we change to left as
if args.checkpointing_steps is not None and global_step % args.checkpointing_steps == 0:

File: examples/stable-diffusion/training/train_text_to_image_sdxl.py (hunk 1 above)
Comment (yeonsily) [reply examples/stable-diffusion/training/train_text_to_image_sdxl.py:68]: done

File: examples/stable-diffusion/training/train_text_to_image_sdxl.py (hunk 3 above)
Comment (yeonsily) [reply examples/stable-diffusion/training/train_text_to_image_sdxl.py:1065]: done

File: examples/stable-diffusion/training/train_text_to_image_sdxl.py (hunk 4 above)
Comment (yeonsily) [reply examples/stable-diffusion/training/train_text_to_image_sdxl.py:1198]: From the original change, checkpointing_steps default set to None.
But ours is 500. Do you want to change default value and add the same condition check here?

File: examples/stable-diffusion/training/train_text_to_image_sdxl.py (hunk 2 above)
Comment (yeonsily) [reply examples/stable-diffusion/training/train_text_to_image_sdxl.py:1151]: not clear which portion you meant. can you please specify more details?

**Comment by libinta** [comment]
No, I mean add
if args.checkpointing_steps is not None

**Comment by libinta** [comment]
If args.use_ema
Is the portion.
Even we didn't enable it. But final PR we should have it uncomment it out.

**Review by skaulintel — APPROVED** [review]

**Review by libinta — APPROVED** [review]

File: examples/stable-diffusion/training/train_text_to_image_sdxl.py [hunk 5]
```diff
@@ -13,12 +13,7 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
-"""
-Fine-tuning script for Stable Diffusion models for text2image with support for LoRA.
-Adapted from the following sources:
-https://github.com/huggingface/diffusers/blob/v0.23.1/examples/text_to_image/train_text_to_image_sdxl.py
```
Comment (libinta) [inline examples/stable-diffusion/training/train_text_to_image_sdxl.py:19]: can you add this to right?

Comment (yeonsily) [reply examples/stable-diffusion/training/train_text_to_image_sdxl.py:19]: You mean adding this comment to our script change? But we don't have any of lora change. We overwrote the original script.

File: examples/stable-diffusion/training/README.md [hunk 6]
```diff
@@ -127,85 +127,63 @@ pip install -r requirements.txt
 ```
 
 ### Example for SDXL
-We can launch the fine-tuning of SDXL model using:
+1x fine-tuning of SDXL model using:
```
Comment (libinta) [inline examples/stable-diffusion/training/README.md:130]: can you follow other examples for 1x/8x wording

File: examples/stable-diffusion/training/gpu/train_text_to_image_sdxl.py [hunk 7]
```diff
@@ -1179,16 +1179,16 @@ def compute_time_ids(original_size, crops_coords_top_left):
...
-                accelerator.log({"train_loss": train_loss_scalar}, step=global_step)
-
-                if args.gradient_accumulation_steps > 1:
-                    logs = {"step_loss": loss.item(), "lr": lr_scheduler.get_last_lr()[0], "mem_used":torch.cuda.memory.memory_allocated()}
-                else:
-                    logs = {"step_loss": train_loss_scalar, "lr": lr_scheduler.get_last_lr()[0], "mem_used":torch.cuda.memory.memory_allocated()}
-                progress_bar.set_postfix(**logs)
-            train_loss.zero_()
+                if (global_step - 1) % args.logging_step == 0 or global_step == args.max_train_steps: #YSY
+                    train_loss_scalar = train_loss.item()
```
Comment (libinta) [inline examples/stable-diffusion/training/gpu/train_text_to_image_sdxl.py:1183]: remove #YSY

File: examples/stable-diffusion/training/run_1x_gaudi1.sh [hunk 8]
```diff
@@ -0,0 +1,19 @@
+PT_HPU_MAX_COMPOUND_OP_SIZE=5 python train_text_to_image_sdxl.py \
```
Comment (libinta) [inline examples/stable-diffusion/training/run_1x_gaudi1.sh:1]: add this to readme as gaudi1 cmd

**Review by libinta — APPROVED** [review]
